src/transformer/events/mod_forum/post_created.php

```diff
@@ -0,0 +1,61 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+namespace src\transformer\events\mod_forum;
+
+defined('MOODLE_INTERNAL') || die();
+
+use src\transformer\utils as utils;
+
+function post_created(array $config, \stdClass $event) {
+    $repo = $config['repo'];
+    $user = $repo->read_record_by_id('user', $event->userid);
+    $course = $repo->read_record_by_id('course', $event->courseid);
+    $post = $repo->read_record_by_id('forum_posts', $event->objectid);
+    $other = unserialize($event->other);
+    $discussionid = $other['discussionid'];
+    $discussion = $repo->read_record_by_id('forum_discussions', $discussionid);
+
+    $lang = utils\get_course_lang($course);
+
+    return[[
+        'actor' => utils\get_user($config, $user),
+        'verb' => [
+            'id' => 'http://id.tincanapi.com/verb/replied',
+            'display' => [
+                $lang => 'replied'
+            ],
+        ],
+        'object' => utils\get_activity\forum_discussion_post($config, $course, $discussionid, $post),
```

Comment:
I was wondering about this and if there was a privacy/GDPR requirement to not do it. If you're doing it at ht2, I think that's good enough evidence that it's appropriate. I'll work on getting that info into the statement.

Reply:
Actually, now that I'm looking at it, it almost seems wrong to do this. Hear me out...

<Actor>, <verb>, <object>.

<User> <replied> <"actual text from the reply message"> is what makes sense to me. Yet that seems like it could get out of hand including the entire text.

If we go with your suggestion, wouldn't the object then simply be the discussion:
```
"object": {
            "id": "http:\/\/www.example.org\/mod\/forum\/discuss.php?d=1",
            "definition": {
                "type": "http:\/\/id.tincanapi.com\/activitytype\/discussion",
                "name": {
                    "en": "test_forum_discussion_name"
                }
            }
        }
```

At that point, we've lost the id (URL) to the actual post (reply) that the student made.

Thoughts?

Reply:
Yeah so I propose `<User> <replied to> <Discussion>`. You don't lose the ID (URL) to the actual reply because you put that into the `statement.context.contextActivities.other`. This is what we do in our LXP when users comment on learning experiences. The `statement.result.response` will then store the actual post text.

Reply:
Gotcha. I'll proceed with that.